### DISCUSSION TITLE
Help registering user

### DISCUSSION THREAD

**Discussion by Mazpaz** [immich-app/immich#131] [Q&A]
Hi I don't have much knowledge of coding, I have downloaded the app on my iPhone but I don't know how to sign up?. if you could please help me with that thanks.

I tried to use terminal on my MacBook Pro and copied the command that was provided for registration but it gives an error curl: (6) Could not resolve host: your-server-ip

I changed the command and put the full IP address but it still doesn't work.

If anyone can help that will be really appreciated

**Comment by alextran1502** [comment]
Hi, were you able to setup and run the server with Docker? or you just downloaded the app?

**Reply by Mazpaz** [reply]
I just download the app. I have no knowledge of Dockers. If you could help me that will be great

**Reply by alextran1502** [reply]
Sure, please follow the step below

1. Follow [this page](https://docs.docker.com/desktop/mac/install/) and install docker on macOS? 
2. Download the Immich repository to your computer.
3. Go into `docker` folder, copy the file `.env.example` and rename to `.env`
4. You can put `immich` for `DB_DATABASE_NAME`, the upload location is the path to a folder on your machine that you want the videos/images to be uploaded. `JWT_SECRET` can be a random text. Put `false` for `ENABLE_MAPBOX` so essentially, it would look like this

```
# Database
DB_USERNAME=postgres
DB_PASSWORD=postgres
DB_DATABASE_NAME=immich

# Upload File Config
UPLOAD_LOCATION=absolute_location_on_your_machine_where_you_want_to_store_the_backup

# JWT SECRET
JWT_SECRET=randomstringthatissolongandpowerfulthatnoonecanguess

# MAPBOX
## ENABLE_MAPBOX is either true or false -> if true, you have to provide MAPBOX_KEY
ENABLE_MAPBOX=false
MAPBOX_KEY=
```

Then, from the terminal, navigate to the docker folder and run `docker-compose up -d`.

When everything is running, you can register a new user by copying and pasting the command below

```bash
curl --location --request POST 'http://localhost:2283/auth/signUp' \
--header 'Content-Type: application/json' \
--data-raw '{
    "email": "testuser@email.com",
    "password": "password"
}'
```

Let me know how it goes.

**Reply by Mazpaz** [reply]
Thanks for this. I can't find the repository?

**Reply by alextran1502** [reply]
The repo means this github project https://github.com/alextran1502/immich

**Comment by ButT00** [comment]
curl --location --request POST 'http://localhost:2283/auth/signUp' \
--header 'Content-Type: application/json' \
--data-raw '{
    "email": "testuser@email.com",
    "password": "password"
}'   Now this not work
is there any other command which work now please tell me thanks

**Reply by bo0tzz** [reply]
https://immich.app/docs/api/create-user-admin
